log4j-api-kotlin/src/test/kotlin/org.apache.logging.log4j.kotlin/LoggerTest.kt

```diff
@@ -173,7 +173,8 @@ class LoggerTest {
   fun `Run in trace with result`() {
     var count = 0
     val f = withLevelFixture(Level.INFO, true) {
-      it.runInTrace(entryMsg) {
+      @SuppressWarnings("unused")
+      val mustBeHere = it.runInTrace(entryMsg) {
```

Comment:
This is not a nice solution, but I could honestly not figure out why this lambda returned `Unit` inside `runInTrace` when the return value was not stored. So to make the test pass I needed to store the return value (which is `1` , the same as it returns inside `runInTrace`).

Looks like an undocumented (or at least very subtle, I tried to find any documentation) breaking change?

Reply:
Hmm, you're right -- seems like a subtle change in behavior due to the new type inference.

We might as well assert on the returned count:

```
    var returnedCount = 0
    val f = withLevelFixture(Level.INFO, true) {
      returnedCount = it.runInTrace(entryMsg) {
        ++count
      }
    }
    assertTrue { count == 1 }
    assertTrue { returnedCount == 1 }
```